### PR TITLE
refactor: dev-server

### PR DESCRIPTION
**Pull request by GiveMe-A-Name**
## Summary
- refactor dev-server's mock middleware
- `@modern-js/app-tools` will createDevServer from '@modern-js/server' instead of 'uni-builder'
- tools.devServer after, before, `setupMiddlewares` will apply mock middleware before and after

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/server/server/src/helpers/mock.ts [hunk 1]
```diff
@@ -67,64 +76,93 @@ export const registerMockHandlers = async ({
...
+    return false;
+  });
+
+  return matched;
+};
+
+export async function initOrUpdateMockMiddlewares(pwd: string) {
+  const mockModule = await getMockModule(pwd);
+
+  const mockHandlers = mockModule?.default;
```
Comment (yimingjfe) [inline packages/server/server/src/helpers/mock.ts:126]: In the getMockModule function, it also require `module.default`, so does this need to get default property twice?

File: packages/server/server/src/helpers/mock.ts [hunk 2]
```diff
@@ -67,64 +76,93 @@ export const registerMockHandlers = async ({
...
   }
 
-  if (!mockModule) {
+  if (!mockHandlers) {
     throw new Error(`Mock file ${mockFilePath} parsed failed!`);
   }
 
-  Object.entries(mockModule as MockConfig).forEach(([key, handler]) => {
+  return {
+    default: mockHandlers,
```
Comment (yimingjfe) [inline packages/server/server/src/helpers/mock.ts:100]: Maybe `mockHandlers` is better than `default`?

**Review by yimingjfe — APPROVED** [review]

File: packages/server/server/src/helpers/mock.ts (hunk 2 above)
Comment (GiveMe-A-Name) [reply packages/server/server/src/helpers/mock.ts:100]: Yes, Great idea

Comment (GiveMe-A-Name) [reply packages/server/server/src/helpers/mock.ts]: Done～

**Review by yimingjfe — APPROVED** [review]

**Pull request merged**